src/snmalloc/backend/meta_protected_range.h

```diff
@@ -0,0 +1,71 @@
+#pragma once
+
+#include "../backend/backend.h"
+
+namespace snmalloc
+{
+  template<
+    typename PAL,
+    typename Pagemap,
+    typename Base,
+    size_t MinSizeBits = MinBaseSizeBits<PAL>()>
+  struct MetaProtectedRangeLocalState
+  {
+  private:
+    // Global range of memory
+    using GlobalR = Pipe<
+      Base,
+      LargeBuddyRange<24, bits::BITS - 1, Pagemap, MinSizeBits>,
+      LogRange<2>,
+      GlobalRange<>>;
+
+    // Central source of object-range, does not pass back to GlobalR as
+    // that would allow flows from Objects to Meta-data, and thus UAF
+    // would be able to corrupt meta-data.
+    using CentralObjectRange = Pipe<
+      GlobalR,
+      LargeBuddyRange<24, bits::BITS - 1, Pagemap>,
+      LogRange<3>,

```

Comment:
What is the number here?

Reply:
I really wanted a string, but I couldn't make that work, so a number was a poor mans string.

Reply:
I see.  It's fairly easy with C++20, so we can hopefully fix that soon.